js_task2/src/js/helpers.js

```diff
@@ -0,0 +1,95 @@
+import { renderRow } from './renderFunctions';
+import {
+  checkedItems,
+  userTemplate,
+  users,
+  numOfRows,
+  outputArea,
+  paginationBox,
+  currentPage,
+} from './_variables';
+
+export function setDataToObject(nodeList) {
+  let i = 0;
+  let object = {};
+  Object.assign(object, userTemplate);
+  for (const key in userTemplate) {
+    object[key] = nodeList[i].textContent;
+    i++;
+  }
+  return object;
+}
+
+export const isCellValid = (cell, fieldValue) => {
+  if (fieldValue === '') {
+    return true;
+  }
+  if (cell.textContent.toLowerCase().indexOf(fieldValue.toLowerCase()) === -1) {
+    return false;
+  }
+
+  return true;
+};
+
+export function selectWholeTable(state) {
+  let rows = document.querySelectorAll('.user-row');
+
+  rows.forEach((row) => {
+    row.firstChild.firstChild.checked = state;
+    if (state) {
+      checkedItems.push(row);
+    } else {
+      checkedItems.pop(row);
+    }
+  });
+}
+
+export const changePage = (pageIndex, btn) => {
```

Comment:
pagination work's a little incorrectly. when you are on the first page the left button should be disabled, the same behavior for the last page and right button. And add please selected-page class to the first page in advance, cause when you are at the beginning of the table, it assumes that you are at the first page

Reply:
Done

Reply:
![image](https://user-images.githubusercontent.com/84564899/127389050-30e2d767-8c3b-4e83-bbc8-e75d3a3fda3d.png)
still reproduced. There is no active first page when loaded

Reply:
![image](https://user-images.githubusercontent.com/84564899/127390174-6d39cc30-441e-413a-92b8-9f676f2bccca.png)
left and right nav arrows should not take selected-page class, they can't be active themselves, only navigate to a page.

Reply:
and now the first page is empty 
![image](https://user-images.githubusercontent.com/84564899/127391375-4524e45e-9db9-400d-aecf-ccd84ab590a4.png)

Reply:
something went wrong from my side, approving this PR according to video confirmation that functionality works correctly